### PR TITLE
feat: Change the way to list the available customer coupons. 

### LINKED ISSUE #6517: Update the customer coupon feature to support the a11y problem
Please refer to: https://app.zeplin.io/project/5c9ddbcf16d4ed776bceeeaa/screen/5e3d71c1d5d27490deceb2c1

### PR DESCRIPTION
**Pull request by Terry01Tan**
close #6517

### CODE REVIEW

**Comment by Jerify02** [comment]
Maybe should run `yarn prettier:fix` once.

**Comment by Terry01Tan** [comment]
> Maybe should run `yarn prettier:fix` once.
fixed

File: projects/storefrontlib/src/cms-components/cart/cart-coupon/cart-coupon.component.html [hunk 1]
```diff
@@ -3,59 +3,7 @@
...
-            >
-              {{ 'voucher.apply' | cxTranslate }}
-            </button>
-          </div>
-        </div>
-      </form>
-    </ng-container>
-
-    <ng-container *cxFeatureLevel="'!1.5'">
+    <ng-container>
```
Comment (kacperknapik) [inline projects/storefrontlib/src/cms-components/cart/cart-coupon/cart-coupon.component.html]: do we still need this `<ng-container>` ?

Comment (Terry01Tan) [reply projects/storefrontlib/src/cms-components/cart/cart-coupon/cart-coupon.component.html]: Deleted.

File: projects/storefrontlib/src/cms-components/cart/cart-coupon/cart-coupon.component.html [hunk 2]
```diff
@@ -3,83 +3,29 @@
...
-            >
-              {{ 'voucher.apply' | cxTranslate }}
-            </button>
-          </div>
+        <div class="col-md-4">
+          <button
+            class="btn btn-block btn-action apply-coupon-button"
+            type="submit"
+            [disabled]="submitDisabled$ | async"
+            [class.disabled]="submitDisabled$ | async"
```
Comment (kacperknapik) [inline projects/storefrontlib/src/cms-components/cart/cart-coupon/cart-coupon.component.html]: According to our latest forms refactors, in general, submit buttons shouldn't be disabled. You should be able to click them, but not submit the form, instead you should show an error on input. Please check #6742

Comment (kacperknapik) [reply projects/storefrontlib/src/cms-components/cart/cart-coupon/cart-coupon.component.html]: Once the button is not `[disabled]` by default, you'll also have to update a11y tests (new interactable element)

File: projects/storefrontlib/src/cms-components/cart/cart-coupon/cart-coupon.component.html [hunk 3]
```diff
@@ -88,4 +34,30 @@
...
+        {{ 'voucher.availableCoupons' | cxTranslate }}
+      </div>
+      <div class="message">
+        {{ 'voucher.availableCouponsLabel' | cxTranslate }}
+      </div>
+      <div class="scroll">
+        <div
+          class="coupons card "
+          *ngFor="let coupon of applicableCoupons"
+          (click)="applyCustomerCoupon(coupon.couponId)"
```
Comment (kacperknapik) [inline projects/storefrontlib/src/cms-components/cart/cart-coupon/cart-coupon.component.html]: `<div>` shouldn't have `click()` event. please move this logic to the button

File: projects/storefrontstyles/scss/components/cart/_cart-coupon.scss [hunk 4]
```diff
@@ -1,68 +1,65 @@
...
 
-    box-shadow: 0 2px 4px -1px rgba(0, 0, 0, 0.2),
-      0 4px 5px 0 rgba(0, 0, 0, 0.14), 0 1px 10px 0 rgba(0, 0, 0, 0.12);
-    border-bottom-left-radius: 4px;
-    border-bottom-right-radius: 4px;
-    margin-top: -0.25rem;
+  .cx-available-coupon {
+    .title {
+      @include type('5');
+      margin: 1.125rem 0 0.5rem;
```
Comment (kacperknapik) [inline projects/storefrontstyles/scss/components/cart/_cart-coupon.scss]: We probably want to use `var()` here too. As seen in the below code.

File: projects/storefrontstyles/scss/components/cart/_cart-coupon.scss [hunk 5]
```diff
@@ -1,68 +1,65 @@
...
-    z-index: 100;
-    position: absolute;
-    max-height: 130px;
-    overflow: auto;
-    width: 100%;
-    background-color: var(--cx-color-inverse);
-    @include type();
+  .cx-cart-coupon-title {
+    @include type('5');
+    margin: 1.125rem 0 0.5rem;
```
Comment (kacperknapik) [inline projects/storefrontstyles/scss/components/cart/_cart-coupon.scss:18]: We probably want to use `var()` here too. As seen in the below code.

File: projects/storefrontstyles/scss/components/cart/_cart-coupon.scss (hunk 4 above)
Comment (Terry01Tan) [reply projects/storefrontstyles/scss/components/cart/_cart-coupon.scss]: fix

File: projects/storefrontstyles/scss/components/cart/_cart-coupon.scss (hunk 5 above)
Comment (Terry01Tan) [reply projects/storefrontstyles/scss/components/cart/_cart-coupon.scss:18]: fix

File: projects/storefrontlib/src/cms-components/cart/cart-coupon/cart-coupon.component.html (hunk 3 above)
Comment (Terry01Tan) [reply projects/storefrontlib/src/cms-components/cart/cart-coupon/cart-coupon.component.html]: fix

File: projects/storefrontlib/src/cms-components/cart/cart-coupon/cart-coupon.component.html (hunk 2 above)
Comment (Terry01Tan) [reply projects/storefrontlib/src/cms-components/cart/cart-coupon/cart-coupon.component.html]: update.

File: projects/storefrontlib/src/cms-components/cart/cart-coupon/cart-coupon.component.ts [hunk 6]
```diff
@@ -174,32 +158,23 @@ export class CartCouponComponent implements OnInit, OnDestroy {
...
 
   applyVoucher(): void {
-    this.cartVoucherService.addVoucher(this.form.value.couponCode, this.cartId);
+    if (this.form.valid) {
+      this.cartVoucherService.addVoucher(
+        this.form.value.couponCode,
+        this.cartId
+      );
+    } else {
+      this.markFormAsTouched();
```
Comment (tobi-or-not-tobi) [inline projects/storefrontlib/src/cms-components/cart/cart-coupon/cart-coupon.component.ts]: Should be simplified with `this.form.markAsTouched();`

Comment (tobi-or-not-tobi) [reply projects/storefrontlib/src/cms-components/cart/cart-coupon/cart-coupon.component.ts]:
```typescript
applyVoucher(): void {
  if (!this.form.valid) {
      this.markFormAsTouched();
      return;
  }
  this.cartVoucherService.addVoucher(
      this.form.value.couponCode,
      this.cartId
    );
}
```

**Pull request merged**